crates/samples/echomain-stateful2/src/test.rs

```diff
@@ -60,14 +60,23 @@ async fn test_partition_info() {
         .unwrap()
         .iter()
         .collect::<Vec<_>>();
-    {
-        assert_eq!(replicas.len(), 3);
-        let replica = &replicas[0];
-        let stateful = match replica {
-            ServiceReplicaQueryResult::Stateful(s) => s,
-            _ => panic!("not stateful"),
-        };
-        assert_eq!(stateful.replica_status, QueryServiceReplicaStatus::Ready);
-        assert_ne!(stateful.node_name, HSTRING::new());
-    }
+    assert_eq!(replicas.len(), 3);
+    let replica = &replicas[0];
+    let stateful_replica = match replica {
+        ServiceReplicaQueryResult::Stateful(s) => s,
+        _ => panic!("not stateful"),
+    };
+    assert_eq!(
+        stateful_replica.replica_status,
+        QueryServiceReplicaStatus::Ready
+    );
+    assert_ne!(stateful_replica.node_name, HSTRING::new());
+
+    let desc = RestartReplicaDescription {
+        node_name: stateful_replica.node_name.clone(),
+        partition_id: single.id,
+        replica_or_instance_id: stateful_replica.replica_id,
+    };
+    let mgmt = fc.get_service_manager();
+    mgmt.restart_replica(&desc, timeout).await.unwrap();
 }
```

Comment:
should we have assert after restart replica

Reply:
Once we have more apis ready I will add checks. The checks here needed is for query service health and status, which is not implemented yet.